distributed/comm/tests/test_comms.py

```diff
@@ -289,6 +289,41 @@ def client_communicate(key, delay=0):
     assert set(l) == {1234} | set(range(N))
 
 
+@gen_test()
+def test_comm_failure_threading():
+    """
+    When we fail to connect, make sure we don't make a lot
+    of threads.
+    """
+
+    @gen.coroutine
+    def sleep_for_500ms():
+        max_thread_count = 0
+        for x in range(500):
+            yield gen.sleep(0.001)
+            thread_count = threading.active_count()
+            if thread_count > max_thread_count:
+                max_thread_count = thread_count
+        return max_thread_count
+
+    # tcp.TCPConnector()
+    sleep_future = gen.convert_yielded(sleep_for_500ms())
```

Comment:
I *think* that you can drop the `gen.convert_yielded` here.  Tornado semantics are a bit different from normal async-def semantics.  `gen.coroutine` objects start running immediately once called.

This is also true below with the `future_connect` lines.  I would merge them into a single `yield connect(...)` statement.